framework/audio/engine/internal/audioengine.cpp

```diff
@@ -117,39 +117,100 @@ IAudioEngine::handle AudioEngine::play(std::shared_ptr<IAudioSource> s, float vo
     }
 
     handle h = m_sl->engine.play(*sa, volume, pan, paused);
+
+    pushMeta(h);
```

Comment:
It's not very clear why I should push some meta, from this context. And what is the meta actually.
For example, the ordering of calls. What would happen if we call it after this block:
`    if (paused) {
        onPause(h);
    } else {
        onPlay(h);
    }`

Reply:
Meta is handler-related data, namely status, status notification channel, playing context and playing context notification channel.
Hendler is created in this place (`handle h = m_sl->engine.play`)
Therefore, in this place, we create the appropriate meta.

`onPause(h)` and other used meta, for example for changing the status

Reply:
gotchya